client/packages/common/src/ui/forms/JsonForms/api/hooks/document/useDocument.ts

```diff
@@ -0,0 +1,18 @@
+import { useQuery } from '@openmsupply-client/common';
+import { useDocumentApi } from '../utils/useDocumentApi';
+
+export const useDocument = (docName: string) => {
+  const api = useDocumentApi();
+
+  return useQuery(
+    api.keys.detail(docName),
+    () => api.get.byDocName(docName),
+    // Don't refetch when the edit modal opens, for example. But, don't cache
+    // data when this query is inactive. For example, when navigating away from
+    // the page and back again, refetch.
+    {
+      refetchOnMount: false,
+      cacheTime: 0,
```

Comment:
think using `staleTime: Infinity` will be helpful. Should prevent background refetches (might want to double check refetching on window focus which is disabled in dev in `host`.. gotcha!) which will result in overriding the current draft when the state (`data`) is synced in `useJSONForms` 😁